addons/mail/static/src/models/message/message.js

```diff
@@ -454,6 +409,25 @@ function factory(dependencies) {
             return parseAndTransform(prettyBody, addLink);
         }
 
+        /**
+         * @private
+         * @returns {mail.thread[]}
+         */
+        _computeThreads() {
+            const cmd = [['replace', []]];
+            cmd.push(['link', this._threads]);
+            if (this.originThread) {
+                cmd.push(['link', this.originThread]);
+            }
+            cmd.push([this.isHistory ? 'link' : 'unlink', this.env.messaging.history]);
+            cmd.push([this.isNeedaction ? 'link' : 'unlink', this.env.messaging.inbox]);
+            cmd.push([this.isStarred ? 'link' : 'unlink', this.env.messaging.starred]);
+            if (this.env.messaging.moderation) {
+                cmd.push([this.isModeratedByCurrentPartner ? 'link': 'unlink', this.env.messaging.moderation]);
+            }
+            return cmd;
+        }
```

Comment:
empty replace -> unlink-all

But the best would be to reorganize the compute to progressively build the final array, and only do one single replace at the end.

So when replace is actually not doing unlink-all internally, we avoid useless unliking/linking.

Reply:
This suggestion wasn't applied or commented.